runtime/src/main/java/org/corfudb/runtime/view/RuntimeLayout.java

```diff
@@ -122,6 +122,14 @@ private IClient getClient(final Class<? extends IClient> clientClass,
             });
             return endpointClientMap;
         }).get(endpoint);
+
+        //Thread priority overrides Client Priority in a TransactionalContext
+        if (TransactionalContext.isInTransaction() &&
+                TransactionalContext.getCurrentContext().getPriorityLevel() != null) {
+            returnClient.setPriorityLevel(TransactionalContext.getRootContext().getPriorityLevel());
+        }
```

Comment:
I think that this is error-prone. Each view maintains a `RuntimeLayout` which is invalidated only when the view encounters a newer layout. In the mean time, subsequent non-transactional operations that go through `returnClient` will use the previously overridden priority level.

Reply:
while we should fix this, we don't have any non-transactional CRUD operations anymore right?

Reply:
Removing this right now since client checkpointing is not going to be enabled now